homeassistant/components/just_nimbus/sensor.py

```diff
@@ -0,0 +1,180 @@
+    ),
+    JustNimbusEntityDescription(
+        key="total_saved",
+        name="Total Saved",
+        icon="mdi:water-opacity",
+        native_unit_of_measurement=VOLUME_LITERS,
+        unique_id="jn:total_saved",
+    ),
+    JustNimbusEntityDescription(
+        key="total_replenished",
+        name="Total Replenished",
+        icon="mdi:water",
+        native_unit_of_measurement=VOLUME_LITERS,
+        unique_id="jn:total_replenished",
+    ),
+    JustNimbusEntityDescription(
+        key="error_code",
+        name="Error Code",
+        icon="mdi:bug",
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement="",
+        unique_id="jn:error_code",
+    ),
+    JustNimbusEntityDescription(
+        key="totver",
+        name="Total Use",
+        icon="mdi:chart-donut",
+        native_unit_of_measurement=VOLUME_LITERS,
+        unique_id="jn:totver",
+    ),
+    JustNimbusEntityDescription(
+        key="reservoir_content_max",
+        name="Max Reservoir Content",
+        icon="mdi:waves",
+        native_unit_of_measurement=VOLUME_LITERS,
+        unique_id="jn:reservoir_content_max",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the JustNimbus sensor."""
+    coordinator: JustNimbusCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        JustNimbusSensor(
+            client=justnimbus.JustNimbusClient(entry.data[CONF_CLIENT_ID]),
+            entry_id=entry.entry_id,
+            device_id=entry.data[CONF_CLIENT_ID],
+            description=description,
+            coordinator=coordinator,
+        )
+        for description in SENSOR_TYPES
+    )
+
+
+class JustNimbusSensor(
+    JustNimbusEntity,
+):
+    """Implementation of the JustNimbus sensor."""
+
+    def __init__(
+        self,
+        *,
+        client: justnimbus.JustNimbusClient,
+        entry_id: str,
+        device_id: str,
+        description: JustNimbusEntityDescription,
+        coordinator: JustNimbusCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        self.entity_description = description
+        super().__init__(
+            client=client,
+            entry_id=entry_id,
+            device_id=device_id,
+            coordinator=coordinator,
+        )
+        self.client = client
+        self._entry_id = entry_id
+        self._device_id = device_id
+        self._attr_unique_id = f"{entry_id}_{description.unique_id}"
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return (
+            super().available
+            and getattr(self.coordinator.data, self.entity_description.key) is not None
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Return sensor state."""
+        return getattr(self.coordinator.data, self.entity_description.key)
```

Comment:
Instead of using `getattr`, consider adding lambdas to the entity descriptions, this keeps typing working as well. Search the codebase for `value_fn` for examples.